File: 2trabajoConstructores/js/clase abstracta/AnimalConParametros.js
const Animal = require("./Animal.js");

class AnimalConParametros extends Animal {
    constructor(nombre, sonido) {
        super();
        this.nombre = nombre;
        this.sonido = sonido;
    }

    hacerSonido() {
        console.log(`${this.nombre} hace: ${this.sonido}`);
    }
}

module.exports = AnimalConParametros;

File: 2trabajoConstructores/c#/claseNormal/AnimalConParametros.cs
public class AnimalConParametros
{
    public string Nombre { get; set; }
    public string Sonido { get; set; }

    public AnimalConParametros(string nombre, string sonido)
    {
        Nombre = nombre;
        Sonido = sonido;
    }

    public void HacerSonido()
    {
        Console.WriteLine($"{Nombre} hace: {Sonido}");
    }
}
